lib/src/main/java/io/ably/lib/rest/AblyBase.java

```diff
@@ -316,6 +317,15 @@ protected void onAuthUpdated(String token, boolean waitForResponse) throws AblyE
         /* Default is to do nothing. Overridden by subclass. */
     }
 
+    /**
+     * Override this method in AblyRealtime and pass updated token to ConnectionManager
+     * @param token new token
+     */
+    protected Future<AblyException> onAuthUpdatedAsync(String token)  {
+        //this must be overriden by subclass
+        return null;
+    }
```

Comment:
Fair question but this is due to current architecture of the app and I just created a similar method as the method in above. Because `AblyRest` is a concrete class and `AblyRealtime` is the one that's going to be using it, we will have to create an empty method on `AblyRest` with the same name if this is described an abstract. Apparently this method is not even suitable to be a class method, it would be better suited as an interface method and only the classes that implements interface would have that in, but here we are 🤷‍♂️